pkg/types/vsphere/validation/platform.go

```diff
@@ -29,8 +29,11 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 		allErrs = append(allErrs, field.Required(fldPath.Child("defaultDatastore"), "must specify the default datastore"))
 	}
 
-	if strings.ToLower(p.VCenter) != p.VCenter {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must be all lower case"))
+	if len(p.VCenter) != 0 {
+		if err := validate.ClusterName1035(p.VCenter); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("vCenter"), p.VCenter, "must conform to RFC 1035 standard. "+
```

Comment:
I don't think that we need to show this much detail in the error here. The user is not picking a hostname: The user is inputting what should be an existing hostname. This error should say something simpler like
```
must be the hostname of the vCenter
```